### DISCUSSION TITLE
How to remove two columns from 12 Step Meeting List

### DISCUSSION THREAD

**Discussion by whateverx28** [code4recovery/12-step-meeting-list#1448] [Q&A] [answered]
We want to remove two columns, Anywhere, and Region from the 12 Step Plugin (using TSML UI) in our site <https://aarichmond2stg.wpenginepowered.com/meetings/> because we are only one region so the choice is meaningless. I don’t see this issue in the forums. WP version 6.5.4 All plugins, themes, and site are updated to latest versions. Thanks in advance for your help.

**Comment by joshreisner** [comment, marked as answer]
hm, i'd like to understand this problem a little more. 

usually we think of "region" as being a city. i see that google wouldnt consider the city of every meeting in this list to be "richmond, va" for example these meetings are in "south hill, va" https://aarichmond2stg.wpenginepowered.com/meetings/?meeting=5th-tradition-3

adding a region for south hill could be useful for people trying to find a meeting if they were in south hill, for example.

did i at one point work with you on a google sheet? are you still importing from it? i think it would be ideal to try to get the cities populated to help people filter the list.

**Reply by joshreisner** [reply]
if youre still uskng the sheet, id be happy to help adjust it to include a region column that reflects the city name. if youre editing meetings in wordpress now then the best way is to add the regions once in meetings > regions, then when you edit the meeting it should autoselect the right one, if the names match

**Reply by whateverx28** [reply]
Still using the sheet, and I am grateful for the help. I haven't been able to do this. I can add the regions in meetings > regions. 
[Richmond.-.RIC.VA.AA.Mtgs.1.csv](https://github.com/user-attachments/files/15877964/Richmond.-.RIC.VA.AA.Mtgs.1.csv)

**Reply by whateverx28** [reply]
Problem is that the region column is all Richmond.

**Reply by whateverx28** [reply]
I think I must go line by line and identify region by zip code. That's what I will do. I entered ten regions and five districts.

**Reply by joshreisner** [reply]
so the [google sheet we worked on in february](https://docs.google.com/spreadsheets/d/1Cf81EPLy3Ii7GtidaLw0MmPyvVi2rl-pCLuRpg3dtSk/edit?gid=2048875353#gid=2048875353) looks a lot different than the CSV uploaded above - it's got proper values in the `City` column - are you still using that sheet?

i see the legacy database at aarichmond.org does have a 'meeting location' field that should work - for example: https://aarichmond.org/meetings.php?ml=CHESTERFIELD&qu=y&zc=&di=5&dw=&nd=&oc=&tm=

we recommend against using postal codes for regions - most people aren't familiar with them - it would be ideal to find a way to correct the City column to have the true city for the meeting if possible. how's the CSV getting generated? do you want to keep using that method or transition to some other system?

**Reply by whateverx28** [reply]
I understand, so I added and verified the regions row by row, and added the regions (42 of them) to the Meetings > Regions, Attached is the updated csv to which I added the number 2 at the end to distinguish it from the original csv ending in number 1. I did not remove the column City, but I think it is now irrelevant.
[Richmond.-.RIC.VA.AA.Mtgs-2.csv](https://github.com/user-attachments/files/15886994/Richmond.-.RIC.VA.AA.Mtgs-2.csv)

**Reply by joshreisner** [reply]
ah ok looks good! the plugin uses `City` to find the meeting, so it's still important keep that column. i would copy the values over from the `Region` column to the `City` column and then drop the `Region` column, if it were me, since this could help if there's an ambiguous address in the bunch (that might look like a different place in richmond)

**Reply by whateverx28** [reply]
I don't trust my skill enough to attempt this action of copying the values over from the Region column to the CIty column, though I know how to drop a column.

**Reply by whateverx28** [reply]
Is that what they call concentenate? I can look it up.

**Reply by joshreisner** [reply]
no, concatenation is joining two things, i'm just suggesting you copy and paste over your city column. perhaps an easier alternative is to delete your city column, and rename region to city, and then fill in the blanks with Richmond?

i noticed a few meetings at the bottom with telephone info in the address column, that info should go in notes, or, better yet, a new column called Conference Phone. having misc information in the address column is going to confuse the importer.

once you have the data loaded on the website, we recommend using wordpress to edit maintain your data. editing in spreadsheets on your computer and reloading is going to cause problems down the road…

**Reply by whateverx28** [reply]
Thank you. I don't think you mean eliminate the Region column. If I rename Region to City, doesn't that eliminate Region? I saw those meetings at the bottom so thanks for the advice.

**Reply by joshreisner** [reply]
ah, i should have noted: if you eliminate `Region` the importer will fall back to use `City`. you really just need `Region` when you need a different value there. in your case i think it might be easier to manage just the single column, but up to you

**Reply by whateverx28** [reply]
Sorry I am slow to understand. I deleted column City since every entry was Richmond, and renamed column Region to City. Looks good to me. DId I do this right?

**Reply by whateverx28** [reply]
Thank you so much for all this help. If github is still taking donations, I'll donate.

**Reply by joshreisner** [reply]
thanks @whateverx28 ! we do accept contributions at https://code4recovery.org/contribute - that will help pay expenses and send us to service events like ICOAA in las vegas.

but we're also just eager to see your site launch!

**Reply by whateverx28** [reply]
Please just confirm I did this right. I deleted column City since every entry was Richmond, and renamed column Region to City.

**Reply by joshreisner** [reply]
i would probably also fill in the blank cities with Richmond, but up to you. if you leave it blank the location of these online meetings will be Virginia, USA, which is also fine

**Reply by whateverx28** [reply]
OK Done. I must ask you why the file name as all these dots in it, not counting the dot before the file extension? Maybe necessary for the software? I have not seen this before.  **Richmond.-.RIC.VA.AA.Mtgs-2.csv**

**Reply by whateverx28** [reply]
Also I tried to donate before but did not receive a response from info@code4recovery.org when I did not receive a link to donate. I just emailed info again. Hope I can donate

**Reply by whateverx28** [reply]
It's up and running! Thank you a million for your help. https://aarichmond2stg.wpenginepowered.com/meetings/ The site will go live by early next week. I messed up though because column Region says Richmond. I think you meant to add another column CIty but clearly I am confused. In any case it looks much better now and is much easier for members and newcomers to find meetings all over the State of VA.

**Reply by joshreisner** [reply]
hm, so strange. could you try reloading the CSV and select `delete all meetings, locations, groups, districts, and regions`? hopefully that will fix the region column. if not, try also editing any meeting listing and saving changes

**Reply by whateverx28** [reply]
That worked! https://aarichmond2stg.wpenginepowered.com/meetings/ You also get a preview of the almost live site. https://aarichmond2stg.wpenginepowered.com/ Thank you very much.

**Reply by whateverx28** [reply]
When it opens, I want it to default to List and not Map. Is there a way to do this?

**Reply by joshreisner** [reply]
yes, list is the default. the links on your home page and in your navigation are linking to the map view. if you can edit the links and take the `?view=map` off the end of those URLs then it will show the list

**Reply by joshreisner** [reply]
(site looks great though!)

**Reply by whateverx28** [reply]
I take it this is something I should do on the Home page, but I do not know where in the navigation to do this. I take it the fix is not in the meetings software.

**Reply by joshreisner** [reply]
that's right

**Reply by whateverx28** [reply]
Is it editing the slug?

**Reply by joshreisner** [reply]
i don't think so, you're just trying to edit the links. i'm guessing the nav is a menu, maybe go to appearance > customize? not sure

i'm guessing someone set this link intentionally - perhaps they know where to fix it?

**Reply by whateverx28** [reply]
Thank you. I'll find it. However, the people on the web committee don't like seeing Region as one of the columns. It's not a problem for me, but is there any way to change that column heading presumably to City

**Reply by joshreisner** [reply]
yes you can do this by adding the following to your theme's functions.php file:

```php
$tsml_ui_config = [
    'strings' => [
        'en' => [
            'region' => 'City'
        ]
    ]
];
```

**Reply by whateverx28** [reply]
Thank you!

**Comment by cronin4392** [comment]
@joshreisner where can I find documentation on the controls available in `tsml_ui_config`? I tried searching in the docs and on Github and only coming up with this thread. Thanks!

Edit:
I just answered my own question, this can be found in the [TSML UI package](https://github.com/code4recovery/tsml-ui). I didn't realize these were separate packages.
